siliconcompiler/tools/klayout/klayout_show.py

```diff
@@ -3,133 +3,147 @@
+    else:
+        lib_lefs = []
+
+    # Load KLayout technology file
+    tech = pya.Technology()
+    if tech_file and os.path.isfile(tech_file):
+        tech.load(tech_file)
+    layoutOptions = tech.load_layout_options
+
+    lefs = []
+
+    lefs.extend(macro_lefs)
+
+    # Technology LEFs -- these are generally specified in the KLayout tech file, but
+    # we overwrite them with the paths in the manifest we don't have to worry if the
+    # paths in the tech file don't resolve right.
+    lefs.extend(tech_lefs)
+    lefs.extend(lib_lefs)
+
+    # Overwrite LEFs specified in tech file with the LEFs we took from the manifest.
+    layoutOptions.lefdef_config.lef_files = lefs
+
+    # These may be disabled in our KLayout tech file for reasons relating to GDS
+    # export, but for the purposes of viewing we'll hardcode them to True.
+    layoutOptions.lefdef_config.produce_blockages = True
+    layoutOptions.lefdef_config.produce_cell_outlines = True
+    layoutOptions.lefdef_config.produce_obstructions = True
+
+    # Always use LEF geometry even when LEF file contains FOREIGN statement.
+    layoutOptions.lefdef_config.macro_resolution_mode = 1
+
+    app = pya.Application.instance()
+
+    # Opinionated default KLayout configuration
+    # see ~/.klayout/klayoutrc for a list of configuration keys
+
+    # show all cells
+    app.set_config('full-hierarchy-new-cell', 'true')
+    # no tip pop-ups
+    app.set_config('tip-window-hidden', 'only-top-level-shown-by-default=3,editor-mode=4,editor-mode=0')
+    # hide text
+    app.set_config('text-visible', 'false')
+    # dark background
+    app.set_config('background-color', '#212121')
+
+    # Display the file!
+    cell_view = pya.MainWindow.instance().load_layout(input_path, layoutOptions, 0)
+    layout_view = cell_view.view()
+
+    if lyp_path:
+        # Set layer properties -- setting second argument to True ensures things like
+        # KLayout's extra outline, blockage, and obstruction layers appear.
+        layout_view.load_layer_props(lyp_path, True)
+
+    # Hide layers that shouldn't be shown in the current view.
+    for layer in layout_view.each_layer():
+        layer_name = layer.name[ : layer.name.find(' - ')]
+        layer_ldt = layer.name[(layer.name.find(' - ') + len(' - ')) : ]
+        if (layer_name in sc_hide_layers) or (layer_ldt in sc_hide_layers):
+            layer.visible = False
+
+    # If 'screenshot' mode is set, save image and exit.
+    if screenshot:
+        # Save a screenshot. TODO: Get aspect ratio from sc_cfg?
+        # horizontal_resolution = int(schema.get('tool', 'klayout', 'task', task, 'var', 'show_horizontal_resolution', step=step, index=index)[0])
+        # vertical_resolution = int(schema.get('tool', 'klayout', 'task', task, 'var', 'show_vertical_resolution', step=step, index=index)[0])
+        horizontal_resolution = 4096
+        vertical_resolution = 4096
```

Comment:
why not get it from the schema

Reply:
Left over from debugging, my bad - nice catch